IdokladSdk/Validation/Attributes/IdentificationNumberPatchAttribute.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace IdokladSdk.Validation.Attributes
+{
+    public class IdentificationNumberPatchAttribute : ValidationAttribute
+    {
+        private readonly string _hasNoIdentificationNumberProperty;
+
+        public IdentificationNumberPatchAttribute(string hasNoIdentificationNumber)
+        {
+            _hasNoIdentificationNumberProperty = hasNoIdentificationNumber;
+        }
+
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            _ = validationContext ?? throw new ArgumentNullException(nameof(validationContext));
+
+            var identificationNumber = (string)value;
+            var hasNoIdentificationNumber = GetHasNoIdentificationNumber(validationContext);
+
+
+            if (identificationNumber != null && hasNoIdentificationNumber != null)
+            {
+                if (!string.IsNullOrEmpty(identificationNumber) && hasNoIdentificationNumber.Value)
```

Comment:
Řeší se někde i whitespaces? Jakože uživatel zadá jako identifikační číslo třeba "     "?

Reply:
whitespace neprojde, viz unittesty. Uživatel musí zadat nemám ICO nebo něco do ICA vložit